src/components/Welcome.js

```diff
@@ -1,7 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
+import { db } from '../lib/firebase';
 
 export default function Welcome() {
+  const [userInputToken, setUserInputToken] = useState('');
+
+  //references the collection where the items are stored
+  let itemsRef = db.collection('items');
+
+  // Handler function for setting the User Input Token state based on user input
+  const handleChange = (event) => {
+    event.preventDefault();
+    setUserInputToken(event.target.value);
+  };
+
+  //Checks to see if userInputToken exists in db and adds user to existing list
+  const handleJoinList = (event) => {
+    event.preventDefault();
+
+    itemsRef
+      .where('userToken', '==', userInputToken)
+      .limit(1)
+      .get()
+      .then(function (querySnapshot) {
+        // if document is empty/ userToken != userInputToken
+        //console log error message
+        if (querySnapshot.empty) {
+          console.log('User Token Does Not Exist');
+          alert('User Token Does Not Exist. Please Try Again');
+          //if userInputToken exists - set it so localStorage & console log items
+        } else {
+          localStorage.setItem('tcl13-token', userInputToken);
+          querySnapshot.forEach(function (doc) {
+            doc.data();
```

Comment:
What is this line doing? It looks like you intended for this to be console logged, but it is not doing that right now.

Reply:
We had removed a console.log statement and I forgot to delete the line. Will do now! Thanks.

Reply:
I am still not sure why you are going through all the data right here. Does it have a purpose?

Reply:
I noticed that we're not being redirected to the list view after we've joined the list. We can access it by clicking on 'View List' but I believe a redirect (history.push) to the list here will fix that.